src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/ModulePotionSpoof.kt

```diff
@@ -0,0 +1,87 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.module.modules.player
+
+import net.ccbluex.liquidbounce.config.types.ToggleableConfigurable
+import net.ccbluex.liquidbounce.event.events.PlayerTickEvent
+import net.ccbluex.liquidbounce.event.handler
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.ClientModule
+import net.minecraft.entity.effect.StatusEffect
+import net.minecraft.entity.effect.StatusEffectInstance
+import net.minecraft.registry.Registries
+import net.minecraft.registry.entry.RegistryEntry
+import net.minecraft.util.Language
+
+/**
+ * PotionSpoof
+ *
+ * Allows the player to have potion effects without actually having the potion.
+ */
+object ModulePotionSpoof : ClientModule("PotionSpoof", Category.PLAYER) {
+
+    private class StatusEffectConfigurable(
+        val registryEntry: RegistryEntry<StatusEffect>,
+        specifiedLanguage: Map<String, String>,
+    ) : ToggleableConfigurable(
+        parent = this,
+        // Value name (en_us)
+        name = specifiedLanguage.getOrDefault(registryEntry.value().translationKey, "Unknown"),
+        enabled = false,
+    ) {
+        private val level = int("Level", 1, 1..10).onChanged {
+            instance = StatusEffectInstance(registryEntry, 0, it - 1)
+        }
+
+        var instance: StatusEffectInstance = StatusEffectInstance(registryEntry, 0, level.get() - 1)
+            private set
+    }
+
+    private val statusEffectValues = run {
+        /** @see Language.create */
+        val language = Language::class.java.getResourceAsStream("/assets/minecraft/lang/en_us.json").let { stream ->
+            val map = HashMap<String, String>(8192)
+            Language.load(stream, map::put)
+            map
+        }
+
+        Registries.STATUS_EFFECT.streamEntries().map {
+            tree(StatusEffectConfigurable(it, specifiedLanguage = language))
+        }.toList()
+    }
+
+    override fun disable() {
+        for (spoofedEffect in statusEffectValues) {
+            if (spoofedEffect.enabled && player.getStatusEffect(spoofedEffect.registryEntry)?.duration == 0) {
+                player.removeStatusEffect(spoofedEffect.registryEntry)
+            }
+        }
+    }
+
+    @Suppress("unused")
+    private val tickHandler = handler<PlayerTickEvent> {
+        for (configurable in statusEffectValues) {
+            if (configurable.enabled) {
+                player.addStatusEffect(configurable.instance)
+            } else if (player.getStatusEffect(configurable.registryEntry)?.duration == 0) {
+                player.removeStatusEffect(configurable.registryEntry)
+            }
+        }
+    }
```

Comment:
Can this not be done by hooking the getStatusEffect functions, so we never have to actually use `addStatusEffect` or `removeStatusEffect` and potentially avoid removing an existing effect or leaving an faked effect?